source/MaterialXCore/Interface.cpp

```diff
@@ -13,8 +13,104 @@
 namespace MaterialX
 {
 
+namespace
+{
+    bool validSourceSwizzlePattern(const string &type, const string &channels)
+    {
+        // Check to see if the channels match those supported by the provided type
+        std::set<char> supportedChannels;
```

Comment:
Maybe constructing these sets could be done once, as a map from type string to channel set. A static member of the PortElement class, or in anonymous namespace within this file. To avoid having to recreate them every time and avoid all the if-elses.

For longer term we could consider making the shadergen `TypeDesc` class part of core, and extending it with query methods for swizzle-ability.

Reply:
I can make this local static. I ran into problems with static initialization order where there was a static in a anonymous namespace in some other file. Could have also been swizzle related :)